src/mscorlib/shared/System/Text/StringBuilder.Debug.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.Serialization;
+
+namespace System.Text
+{
+    public sealed partial class StringBuilder : ISerializable
+    {
+        private void ShowChunks(int maxChunksToShow = 10)
+        {
+            Debug.WriteLine('|' + string.Join('|', ShowChunksInOrder(maxChunksToShow)) + '|');
+        }
+
+        private IEnumerable<string> ShowChunksInOrder(int maxChunksToShow)
```

Comment:
I didnt end up using yield because I am populating the IEnumerable from last to first.